backend/smb/filepool.go

```diff
@@ -0,0 +1,91 @@
+package smb
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/cloudsoda/go-smb2"
+	"golang.org/x/sync/errgroup"
+)
+
+type smbChunkWriterFile struct {
```

Comment:
This then should be called `poolFile` or maybe `filepoolFile` or maybe just `file`.